### PR TITLE
wr-132:  Add date-fns package for dates

### PR DESCRIPTION
**Pull request by hnoievets**
- [x] Add [date-fns](https://date-fns.org/) package to shared folder
- [x]  Provide helpers for date formatting
- [x]  Replace places throughout the code to use date-fns helpers to manage dates shape

### CODE REVIEW

File: shared/src/libs/enums/date-pattern.emun.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const DatePattern = {
```
Comment (NazarKovtun1) [inline shared/src/libs/enums/date-pattern.emun.ts]: typo in file name

**Review by NazarKovtun1 — CHANGES_REQUESTED** [review]

File: shared/src/libs/enums/date-pattern.emun.ts (hunk 1 above)
Comment (hnoievets) [reply shared/src/libs/enums/date-pattern.emun.ts]: Thanks

File: shared/src/libs/enums/date-pattern.enum.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const DatePattern = {
+  MONTH_DAY: 'd',
+  SHORT_MONTH_NAME: 'LLL',
+} as const;
```
Comment (andanf-e) [inline shared/src/libs/enums/date-pattern.enum.ts]:
```suggestion
const DateFormat = {
  DAY_SHORT_MONTH: 'd MMM',
} as const;
```

File: shared/src/libs/helpers/format-date/format-date.helper.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    useAdditionalDayOfYearTokens?: boolean;
+  };
+};
+
+const formatDate = ({
+  isoString,
+  pattern,
+  options,
+}: FormatDateParameters): string => {
+  return format(new Date(isoString), pattern, options);
```
Comment (andanf-e) [inline shared/src/libs/helpers/format-date/format-date.helper.ts]:
```suggestion
const getFormattedDate = (date: string, format: ValueOf<typeof DateFormat>): string => {
  return format(new Date(date), format);
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

**Review by andanf-e — APPROVED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Pull request merged**